### PR TITLE
Esword cig lighting messages fix

### PR DESCRIPTION
**Pull request by PKPenguin321**
Fixes https://github.com/tgstation/tgstation/issues/37213

### CODE REVIEW

**Comment by Tlaltecuhtli** [comment]
make it gender sensitivie

File: code/game/objects/items/melee/energy.dm [hunk 1]
```diff
@@ -55,7 +55,7 @@
 		var/mob/living/carbon/C = user
 		if(C.wear_mask == src)
```
Comment (SpaceManiac) [inline code/game/objects/items/melee/energy.dm]: While we're here I'm pretty sure this check and the one in the other file are wrong, given that these swords cannot be worn on the face.

File: code/game/objects/items/twohanded.dm [hunk 2]
```diff
@@ -417,9 +417,9 @@
 	var/in_mouth = ""
 	if(iscarbon(user))
 		var/mob/living/carbon/C = user
-		if(C.wear_mask == src)
+		if(C.wear_mask)
 			in_mouth = ", barely missing their nose"
-	. = "<span class='warning'>[user] swings [user.p_their()] [src][in_mouth]. [user.p_they()] light[user.p_s()] [A] in the process.</span>"
+	. = "<span class='warning'>[user] swings [user.p_their()] [name][in_mouth]. They light their [A.name] in the process.</span>"
```
Comment (ShizCalev) [inline code/game/objects/items/twohanded.dm]: ` -	. = "<span class='warning'>[user] swings [user.p_their()] [name][in_mouth]. [user.p_they(TRUE)] light[user.p_s()]  [user.p_their()] [A.name] in the process.</span>" `

File: code/game/objects/items/melee/energy.dm [hunk 3]
```diff
@@ -53,9 +53,9 @@
 	var/in_mouth = ""
 	if(iscarbon(user))
 		var/mob/living/carbon/C = user
-		if(C.wear_mask == src)
+		if(C.wear_mask)
 			in_mouth = ", barely missing their nose"
-	. = "<span class='warning'>[user] swings their [src][in_mouth]. They light [A] in the process.</span>"
+	. = "<span class='warning'>[user] swings their [name][in_mouth]. They light their [A.name] in the process.</span>"
```
Comment (ShizCalev) [inline code/game/objects/items/melee/energy.dm]: Mirror it here too.

File: code/game/objects/items/twohanded.dm (hunk 2 above)
Comment (ShizCalev) [reply code/game/objects/items/twohanded.dm]: true for the first arg to `p_they` capitalizes the output.

**Review by ShizCalev — CHANGES_REQUESTED** [review]

**Pull request merged**